File: tests/fixtures/sample.go
// SPDX-License-Identifier: GPL-2.0-or-later
package main

import (
	"fmt"
	"regexp"
)

const defaultLimit = 42

type Worker struct {
	Name   string
	Active bool
}

func (worker Worker) Run(items []string) int {
	pattern := regexp.MustCompile(`^task:\d+$`)
	count := 0
	for _, item := range items {
		if worker.Active && pattern.MatchString(item) {
			count++
		}
	}
	return count
}

func main() {
	fmt.Println(Worker{Name: "limei", Active: true}.Run([]string{"task:1"}))
}

File: tests/fixtures/sample.rs
// SPDX-License-Identifier: GPL-2.0-or-later
const DEFAULT_LIMIT: usize = 42;

struct Worker {
    name: String,
    active: bool,
}

impl Worker {
    fn run(&self, values: &[usize]) -> usize {
        values
            .iter()
            .filter(|value| self.active && **value < DEFAULT_LIMIT)
            .map(|value| value * 2)
            .sum()
    }
}

fn main() {
    let worker = Worker { name: "limei".into(), active: true };
    println!("{}:{}", worker.name, worker.run(&[1, 2, 3]));
}
